crates/biome_parser/src/diagnostic.rs

```diff
@@ -32,22 +34,23 @@ pub struct ParseDiagnostic {
 }
 
 /// Possible details related to the diagnostic
-#[derive(Debug, Default, Clone)]
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
 struct ParserAdvice {
     advice_list: Vec<ParserAdviceKind>,
 }
 
 /// The structure of the advice. A message that gives details, a possible range so
 /// the diagnostic is able to highlight the part of the code we want to explain.
-#[derive(Debug, Clone)]
+#[derive(Clone, Debug, Deserialize, Serialize)]
```

Comment:
These should not be serializable/deserialisable. When and if we require them to be serialisable, we should use `diagnostics::serde`. Here is an example:

https://github.com/biomejs/biome/blob/f6fd1ef7130e7bcd1466aa78ae7295fc5fdd5d58/crates/biome_service/src/file_handlers/json.rs#L350-L362